### PR TITLE
[google-spreadsheet] add missing headerRowIndex options

### PR DESCRIPTION
**Pull request by hoonoh**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/theoephraim/node-google-spreadsheet/commit/198e9c3aa6053f72f8a6e0da0f2019fcaa74f479>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/google-spreadsheet/index.d.ts [hunk 1]
```diff
@@ -763,15 +763,18 @@ export class GoogleSpreadsheetWorksheet implements WorksheetBasicProperties {
      * set the header (first) row in the worksheet
      *
      * @param headers
+     * @param headerRowIndex
```
Comment (EndBug) [inline types/google-spreadsheet/index.d.ts:766]:
```suggestion
     * @param headerRowIndex The index of the header row, if not the first. NOTE: not zero-indexed
```

File: types/google-spreadsheet/index.d.ts [hunk 2]
```diff
@@ -763,15 +763,18 @@ export class GoogleSpreadsheetWorksheet implements WorksheetBasicProperties {
...
      */
-    setHeaderRow(headers: string[]): Promise<void>;
+    setHeaderRow(headers: string[], headerRowIndex?: number): Promise<void>;
 
     /**
      * @description
      * loads the header row (first row) of the sheet
      * - usually do not need to call this directly
+     *
+     * @param headerRowIndex
```
Comment (EndBug) [inline types/google-spreadsheet/index.d.ts:775]:
```suggestion
     * @param headerRowIndex The index of the header row, if not the first. NOTE: not zero-indexed
```

**Review by EndBug — CHANGES_REQUESTED** [review]
Hi, everything looks good! I'd just add a bit of JSDoc since it's not zero-indexed, and it may be confusing otherwise.

**Comment by EndBug** [comment]
Also, related to #57591

**Comment by hoonoh** [comment]
@EndBug added @param description text as you have suggested!

**Review by EndBug — APPROVED** [review]
LGTM ✅

**Comment by hoonoh** [comment]
Ready to merge

**Pull request merged**